### PR TITLE
use rxjs

### PR DESCRIPTION
**Pull request by sharapinski**
1. changed 'course.service.ts' to Observables. 
2. added a filter in 'course.service.ts'
3. done in subtask 2 
4. used replay subject in 'auth.service.ts'
5. skiped???
6. skiped

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
+    this.isAuthorized = this._authService.isAuthorized();
+  }
 
   ngOnInit() {
-    this.isAuthorized = this._authService.isAuthorized();
+    this._authService.subject.subscribe(res => this.isAuthorized = !!res);
+  }
+
+  ngOnDestroy() {
+    this._subscription.unsubscribe();
```
Comment (Antontelesh) [inline src/app/app.component.ts]: and where `this._subscription` is defined?

File: src/app/app.component.ts [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
 import { AuthService } from './shared/auth.service';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html'
 })
-export class AppComponent implements OnInit{
+export class AppComponent implements OnInit, OnDestroy {
   isAuthorized: boolean;
+  _subscription: any;
```
Comment (Antontelesh) [inline src/app/app.component.ts]: `this._subscription` is not of type `any`, but of type `Subscription` which can be imported from `rxjs/Subscription`

File: src/app/shared/course.service.ts [hunk 3]
```diff
@@ -7,38 +15,52 @@ import { Course } from './course';
...
   courses: Course[] = courses;
 
-  getList(): Course[] {
-    return this.courses;
+  getList(): Observable<Course[]> {
+    // clone array to use the same ref
+    // const courses = this.courses.map(x => Object.assign({}, x));
+
+    return Observable.of(courses)
+                    .map(list => list.filter(course => Number(course.date) > new Date().getTime() - 14 * 24 * 60 * 1000))
```
Comment (Antontelesh) [inline src/app/shared/course.service.ts]: please, move this to component that accepts `Observable<Course[]>`

File: src/app/pages/course-list/course-list.component.ts [hunk 4]
```diff
@@ -13,10 +13,16 @@ export class CourseListComponent{
 
   constructor(private _service : CourseService){}
 
-  onItemDelete(item: Course){
+  onDelete(item: Course){
     var result: boolean = confirm("Do you really want to delete this course? ");
     if (result) {
-      this._service.removeItem(item);
+      this._service.removeItem(item).subscribe(res => {
```
Comment (Antontelesh) [inline src/app/pages/course-list/course-list.component.ts]: it's bad to subscribe to an observable than never ends. you introduce memory leaks by doing this.
it's better to call removeItem() and treat this as an effect.
the service should be responsible for emitting a new updated array of courses in the main stream.

**Review by Antontelesh — CHANGES_REQUESTED** [review]
Please, apply the necessary changes and then merge

File: src/app/app.component.ts (hunk 2 above)
Comment (sharapinski) [reply src/app/app.component.ts]: ok

**Pull request closed without merging**